enumerable.rb

```diff
@@ -0,0 +1,139 @@
+# Enumerable method
+# rubocop:disable all
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = self if self.class == Array
+    arr = to_a if self.class == Range
+    arr = flatten if self.class == Hash
+    for i in arr
+      yield i
+    end
+    arr
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    arr = self if self.class == Array
+    arr = to_a if self.class == Range
+    arr = flatten if self.class == Hash
+    for i in 0...(arr.length)
+      yield arr[i], i
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    my_each do |item|
+      result.push(item) if yield(item)
+    end
+    result
+  end
+
+  def my_all?(arg = nil)
+    if block_given?
+      my_each { |item| return false if yield(item) == false }
+      return true
+    elsif arg.nil?
+      my_each { |num| return false if num.nil? || num == false }
+    elsif !arg.nil? && (arg.is_a? Class)
+      my_each { |num| return false if num.class != arg }
+    elsif !arg.nil? && arg.class == Regexp
+      my_each { |num| return false unless arg.match(num) }
+    else
+      my_each { |num| return false if num != arg }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if block_given?
+      my_each { |item| return true if yield(item) }
+      false
+    elsif arg.nil?
+      my_each { |num| return true if num }
+    elsif !arg.nil? && (arg.is_a? Class)
+      my_each { |num| return true if num.class == arg }
+    elsif !arg.nil? && arg.class == Regexp
+      my_each { |num| return true if arg.match(num) }
+    else
+      my_each { |num| return true if num == arg }
+    end
+    false
+  end
+
+  def my_none?(arg = nil)
+    if !block_given? && arg.nil?
+      my_each { |item| return false if yield(item) }
+      true
+    end
+    if !block_given? && !arg.nil?
+      if arg.is_a? Class
+        my_each { |num| return false if num.class == arg }
+        true
+      end
+      if !arg.nil? && arg.class == Regexp
+        my_each { |num| return false if arg.match(num) }
+        true
+      end
+      my_each { |num| return false if num == arg }
+      true
+    end
+    my_any? { |item| return false if yield(item) }
+    true
+  end
```

Comment:
- [ ] when no block or argument is given it should return true only if none of the collection members is true


```Ruby
  false_array =  [nil, false, true, []] 

  false_array.my_none? # should return true, but got LocalJumpError: no block given (yield) instead
```

- [ ] when no block or argument is given it should return false only if one of the collection members is true
    

```Ruby
  true_array =  [nil, false, true, []] 

  true_array .my_none? # should return false, but got LocalJumpError: no block given (yield) instead

```

- [ ] when a class is passed as an argument it should return true if none of the collection is a member of such class

```Ruby

  array.my_none?(String) # should return true, but got LocalJumpError: no block given (yield) instead

```

- [ ] when a class is passed as an argument it should return true if none of the collection is a member of such class::Numeric


```Ruby

  array.my_none?(Numeric) # should return false, but got LocalJumpError: no block given (yield) instead

```

- [ ] when a pattern other than Regex or a Class is given it should return true only if none of the collection matches the pattern

```Ruby

  array.my_none?(/z/) # should return true, but got LocalJumpError: no block given (yield) instead

```

- [ ] when a pattern other than Regex or a Class is given it should return true only if none of the collection matches the pattern

```Ruby

  array.my_none?(5) # should return true, but got LocalJumpError: no block given (yield) instead

```